doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -344,85 +343,94 @@ Rewriting with definitional equality
-   construction inside the :n:`@term` itself (use here the ``zeta`` flag). In
-   any cases, opaque constants are not unfolded (see :ref:`vernac-controlling-the-reduction-strategies`).
-
-   Normalization according to the flags is done by first evaluating the
+-----------------------
+
+.. todo PR: Is the above good title?  Perhaps "Applying reductions" would be better?
+
+These tactics apply reductions, replacing :term:`convertible` terms
+with equivalent terms.  Other ways to apply these reductions are through
+the :cmd:`Eval` command, the `Eval` clause in the :cmd:`Definition`/:cmd:`Example`
+command and the :tacn:`eval` tactic.
+
+.. todo PR: Above seems like a very wimpy introduction.
+   Clearly "compute" does conversions that correspond very closely
+   to the notion of evaluating an expression, but IIUC others are
+   directed at, say, expanding a term so it can be handled by following
+   tactics.
+
+   I didn't see how comparing these to "change" was very helpful, so I
+   dropped that sentence, too.
+
+.. tacn:: lazy {? @strategy_flag } {? @occurrences }
+          cbv {? @strategy_flag } {? @occurrences }
+
+   .. insertprodn red_expr pattern_occs
+
+   .. prodn::
+      red_expr ::= red
+      | hnf
+      | simpl {? @delta_flag } {? {| @reference_occs | @pattern_occs } }
+      | cbv {? @strategy_flag }
+      | cbn {? @strategy_flag }
+      | lazy {? @strategy_flag }
+      | compute {? @delta_flag }
+      | vm_compute {? {| @reference_occs | @pattern_occs } }
+      | native_compute {? {| @reference_occs | @pattern_occs } }
+      | unfold {+, @reference_occs }
+      | fold {+ @one_term }
+      | pattern {+, @pattern_occs }
+      | @ident
+      delta_flag ::= {? - } [ {+ @reference } ]
+      strategy_flag ::= {+ @red_flag }
+      | @delta_flag
+      red_flag ::= beta
+      | iota
+      | match
+      | fix
+      | cofix
+      | zeta
+      | delta {? @delta_flag }
+      reference_occs ::= @reference {? at @occs_nums }
+      pattern_occs ::= @one_term {? at @occs_nums }
+
+   Normalize the goal as specified by the :n:`@strategy_flag`\s. The flags
+   include:
+
+   `beta`
+     :term:`beta-reduction` of functional application
+
+   `delta`
+     :term:`delta-reduction`: unfolding of transparent constants, see :ref:` controlling-the-reduction-strategies`
+
+   :n:`delta {? - } [ {+ @reference } ]`
+     without the `-`, limits unfolding to the listed constants.  If the `-` is present,
+     unfolding is applied to all constants that are not listed.
+     Notice that the ``delta`` flag doesn't apply to variables bound by a let-in
+     construction inside the :n:`@term` itself (use the ``zeta`` flag to unfold these).
+     Opaque constants are never unfolded
+     (see :ref:` controlling-the-reduction-strategies`).
+
+   `iota`
+     :term:`iota-reduction` of pattern matching (`match`) over a constructed term and unfolding
+     of :g:`fix` and :g:`cofix` expressions.  Shorthand for `match fix cofix`
```

Comment:
How is this important to readers?  (I would put the statement and the example in the conversion chapter, BTW.)

Reply:
It's not completely obvious that `match Some 0 with Some x => x | None => ... end` will be iota-reduced to `(fun x : nat => x) 0` (reproducing somehow the selected branch) rather than directly to `0`. So it may be relevant to readers who may want to combine `iota` and `beta`. But yes, this could be said in the conversion chapter (maybe in your subsequent PR).